src/Simulators/MultipleTimedRenders.js

```diff
@@ -0,0 +1,62 @@
+import { createElement, Component } from 'react';
+import { arrayOf, oneOfType, shape, func, number, any } from 'prop-types';
+import SimulatorErrorBoundary from './SimulatorErrorBoundary';
+import scheduleCallbackArgs from './schedule-callback-args';
+
+export default class MultipleTimedRenders extends Component {
+    static propTypes = {
+        initialArgs: oneOfType([arrayOf(any), func]),
+        scheduledArgs: arrayOf(
+            shape({
+                elapsed: number.isRequired,
+                args: oneOfType([arrayOf(any), func]).isRequired
+            }).isRequired
+        ).isRequired,
+        /* (error: Error) => any */
+        onError: func,
+        /* (prop: any) => React.Element<any> */
+        children: func.isRequired
+    };
+
+    constructor(props) {
+        super(props);
+        const { initialArgs } = props;
+        this.state = {
+            args:
+                typeof initialArgs === 'function' ? initialArgs() : initialArgs
+        };
+    }
+
+    componentDidMount() {
+        this._pending = scheduleCallbackArgs(
```

Comment:
I haven't seen many React codebases in the wild using pseudo-privates with an `_`. You should never be constructing a component manually, and almost never grab a ref to a composite, so it's  not super necessary.

[We haven't been using them in Venia at all](https://github.com/magento-research/pwa-module-theme/blob/af13624c3708b5667275ba7e586423d4789cab94/theme/src/view/Gallery/item.js)

Reply:
Would you suggest that instead we maintain the instance variable in `this.state`, and then implement `shouldComponentUpdate` to prevent unnecessary renders?